coderunners/scoring.py

```diff
@@ -0,0 +1,26 @@
+from abc import abstractmethod
+
+from models import Status, TestGroup
+
+
+class AbstractScorer:
+    @abstractmethod
+    def get_score(self, test_results: list[Status], test_scores: list[int]) -> float:
+        raise NotImplementedError()
+    
+class PerTestScorer(AbstractScorer):
+    def get_score(self, test_results: list[Status], test_scores: list[int]) -> float:
+        return sum(test_scores) / len(test_scores)
+
+class SubtaskScorer(AbstractScorer):
```

Comment:
Would be better to have it as `dataclass` to avoid having a redundant `__init__`